backend/runner/runner.go

```diff
@@ -314,36 +304,36 @@ func (s *Service) Deploy(ctx context.Context, req *connect.Request[ftlv1.DeployR
 	)
 	if err != nil {
 		observability.Deployment.Failure(ctx, optional.Some(key.String()))
-		return nil, fmt.Errorf("failed to spawn plugin: %w", err)
+		return fmt.Errorf("failed to spawn plugin: %w", err)
 	}
 
 	dep := s.makeDeployment(cmdCtx, key, deployment)
 	s.deployment.Store(optional.Some(dep))
 
+	logger.Debugf("Deployed %s", key)
 	setState(ftlv1.RunnerState_RUNNER_ASSIGNED)
+	context.AfterFunc(ctx, func() {
+		err := s.Terminate()
+		if err != nil {
+			logger := log.FromContext(ctx)
+			logger.Errorf(err, "failed to terminate deployment")
+		}
+	})
 
-	return connect.NewResponse(&ftlv1.DeployResponse{}), nil
+	return nil
 }
 
-func (s *Service) Terminate(ctx context.Context, c *connect.Request[ftlv1.TerminateRequest]) (*connect.Response[ftlv1.RegisterRunnerRequest], error) {
+func (s *Service) Terminate() error {
```

Comment:
Not a big deal but probably slightly more idiomatic to call this `Close() error`